src/postprocess/chi0.jl

```diff
@@ -103,33 +103,101 @@ LinearAlgebra.ldiv!(y::T, P::FunctionPreconditioner, x) where {T} = P.preconditi
 LinearAlgebra.ldiv!(P::FunctionPreconditioner, x) = (x .= P.precondition!(similar(x), x))
 precondprep!(P::FunctionPreconditioner, ::Any) = P
 
-# Solves Q (H-εn) Q δψn = -Q rhs
-# where Q is the projector on the orthogonal of ψk
-function sternheimer_solver(Hk, ψk, ψnk, εnk, rhs; tol_cg=1e-6, verbose=false)
+# Solves (1-P) (H-εn) (1-P) δψn = - (1-P) rhs
+# where 1-P is the projector on the orthogonal of ψk
+function sternheimer_solver(Hk, ψk, ψnk, εnk, rhs;
+                            ψk_extra=zeros(size(ψk,1), 0),
+                            εk_extra=zeros(0), schur=true,
```

Comment:
Why would you ever turn the Schur off explicitly? You can do so just by passing no extra bands. I would remove that kwarg.